src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/KeyStatisticServiceTests.cs

```diff
@@ -607,6 +765,257 @@ await using (var context = InMemoryContentDbContext(contextId))
+                keyStatisticText,
+            }
+        };
+
+        var contextId = Guid.NewGuid().ToString();
+        await using (var context = InMemoryContentDbContext(contextId))
+        {
+            await context.Releases.AddRangeAsync(release);
+            await context.SaveChangesAsync();
+        }
+
+        await using (var context = InMemoryContentDbContext(contextId))
+        {
+            var keyStatisticService = SetupKeyStatisticService(context);
+            var result = await keyStatisticService.UpdateKeyStatisticText(
+                release.Id,
+                Guid.NewGuid(),
+                new KeyStatisticTextUpdateRequest
+                {
+                    Title = "new title",
+                    Statistic = "new statistic",
+                    Trend = "new trend",
+                    GuidanceTitle = "new guidanceTitle",
+                    GuidanceText = "new guidanceText",
+                });
+
+            result.AssertNotFound();
+        }
+    }
+
+    [Fact]
+    public async Task UpdateKeyStatisticText_KeyStatWrongType()
+    {
+        var releaseId = Guid.NewGuid();
+        var dataBlockId = Guid.NewGuid();
+
+        var keyStatisticDataBlock = new KeyStatisticDataBlock
+        {
+            ReleaseId = releaseId,
+            DataBlockId = dataBlockId,
+            Trend = "trend",
+            GuidanceTitle = "guidanceTitle",
+            GuidanceText = "guidanceText",
+            Created = new DateTime(2000, 1, 1),
+            Updated = null,
+        };
+
+        var release = new Release
+        {
+            Id = releaseId,
+            ContentBlocks = new List<ReleaseContentBlock>
+            {
+                new()
+                {
+                    ContentBlock = new DataBlock
+                    {
+                        Id = dataBlockId,
+                    },
+                },
+            },
+            KeyStatistics = new List<KeyStatistic>
+            {
+                keyStatisticDataBlock,
+            }
+        };
+
+        var contextId = Guid.NewGuid().ToString();
+        await using (var context = InMemoryContentDbContext(contextId))
+        {
+            await context.Releases.AddRangeAsync(release);
+            await context.SaveChangesAsync();
+        }
+
+        await using (var context = InMemoryContentDbContext(contextId))
+        {
+            var keyStatisticService = SetupKeyStatisticService(context);
+            var result = await keyStatisticService.UpdateKeyStatisticText(
+                release.Id,
+                keyStatisticDataBlock.Id,
+                new KeyStatisticTextUpdateRequest
+                {
+                    Title = "new title",
+                    Statistic = "new statistic",
+                    Trend = "new trend",
+                    GuidanceTitle = "new guidanceTitle",
+                    GuidanceText = "new guidanceText",
+                });
+
+            result.AssertNotFound();
```

Comment:
I was almost expecting an `AssertBadRequest` to be here, but I'm not too fussed either way - it does indeed show that the KeyStatText with that ID does not exist!

Reply:
I've left it